qutebrowser/misc/editor.py

```diff
@@ -48,10 +50,12 @@ def __init__(self, parent=None):
         self._filename = None
         self._proc = None
         self._remove_file = None
+        self._watcher = QFileSystemWatcher(parent=self)
 
     def _cleanup(self):
         """Clean up temporary files after the editor closed."""
         assert self._remove_file is not None
+        self._watcher.fileChanged.disconnect()
```

Comment:
This fails in [various tests](https://travis-ci.org/qutebrowser/qutebrowser/jobs/320641816#L3695) - can we guarantee that the signal is always connected when running `_cleanup`? Probably not, right? If we can't, this should be in a `try:` block where we catch `TypeError`.

Also, it'd be better to pass the slot we're disconnecting from here (I try to avoid blanket `.disconnect()` calls because it could be quite unexpected when you add more connections in the future).